src/Cake.Common/Tools/MSBuild/MSBuildVerbosityExtensions.cs

```diff
@@ -35,5 +35,29 @@ public static string GetMSBuildVerbosityName(this Verbosity verbosity)
                     throw new CakeException("Encountered unknown MSBuild build log verbosity.");
             }
         }
+
+        /// <summary>
+        /// Gets the MSBuild <see cref="Verbosity"/> from string value.
+        /// </summary>
+        /// <param name="verbosity">The verbosity string value.</param>
+        /// <returns>MSBuild <see cref="Verbosity"/> enumeration.</returns>
+        public static Verbosity GetMSBuildVerbosity(this string verbosity)
+        {
+            switch (verbosity?.ToLower())
+            {
+                case "quiet":
+                    return Verbosity.Quiet;
+                case "minimal":
+                    return Verbosity.Minimal;
+                case "normal":
+                    return Verbosity.Normal;
+                case "detailed":
+                    return Verbosity.Verbose;
+                case "diagnostic":
+                    return Verbosity.Diagnostic;
+                default:
+                    throw new CakeException("Encountered unknown MSBuild build log verbosity.");
```

Comment:
Perhaps we could output known values here as well.

`Encountered unknown MSBuild build log verbosity 'foo'. Valid values are 'quiet', 'minimal', 'normal', 'detailed' and 'diagnostic'.`